crates/curp-test-utils/src/test_cmd.rs

```diff
@@ -287,28 +287,33 @@ impl CommandExecutor<TestCommand> for TestCE {
     fn after_sync(
         &self,
         cmds: Vec<AfterSyncCmd<'_, TestCommand>>,
-        highest_index: LogIndex,
+        highest_index: Option<LogIndex>,
     ) -> Vec<Result<AfterSyncOk<TestCommand>, <TestCommand as Command>::Error>> {
         let as_duration = cmds
             .iter()
             .fold(Duration::default(), |acc, c| acc + c.cmd().as_dur);
         std::thread::sleep(as_duration);
         let total = cmds.len();
-        for (i, cmd) in cmds.iter().enumerate() {
-            let index = highest_index - (total - i - 1) as u64;
-            self.after_sync_sender
-                .send((cmd.cmd().clone(), index))
-                .expect("failed to send after sync msg");
+        let mut wr_ops = Vec::new();
+
+        if let Some(index) = highest_index {
+            for (i, cmd) in cmds.iter().enumerate() {
+                let index = index - (total - i - 1) as u64;
```

Comment:
What does this formula mean? Is the `hightest_index` a logical index or something like that?

Reply:
The highest_index means the last index of all cmds, this formula is used to calculate the index of the current cmd.